### DISCUSSION TITLE
Display custom 404 page if model cannot be found

### DISCUSSION THREAD

**Discussion by e-m-s-y** [filamentphp/filament#14763] [Help] [answered]
### Package

Other

### Package Version

v3.2.122

### How can we help you?

Hello community,

I currently have a use-case where I would like to view a custom view if a model cannot be found. At the moment the default Laravel 404 view is shown.

**Example**
The app has a Page model which can be queried by name instead of ID by using model route binding. For example, `app.example/page/$name`. If the Page cannot be found I would like to render a page with the option for the user to create the page. Think "Looks like we can't find this page, would you like to create it?", something like that. This only has to work for this particular modal / resource. Other resources can just return the default 404.

I've tried several things but I couldn't get it working, I couldn't find anything in the docs as well.

Things I've tried:
1. Override `resolveRecord` and catch the ModelNotFoundException in order to redirect to a custom page. `resolveRecord` has to return a model resulting in errors. 
2. Instead of redirecting I tried to throw a custom exception in order to render a view, this is Laravel behaviour see https://laravel.com/docs/11.x/errors#renderable-exceptions. I also got into trouble because I couldn't get it working to render a Filament page from within the exception's render function.
3. I tried to add a custom page and put it right below the view page in the `getPages` function, both same path. My idea was a sort of fallback. This didn't work because the ModelNotFoundException just renders the Laravel 404 view from the initial call.
4. Laravel offers customization for missing models but I couldn't find a way to implement this into Filament: https://laravel.com/docs/11.x/controllers#customizing-missing-model-behavior.

TDLR: What's the Filament way to render a custom view if a model can't be found using model route binding?

**Comment by leandrocfe** [comment, marked as answer]
https://github.com/filamentphp/filament/discussions/14769#discussioncomment-11226473
You could use the same idea and show a custom page instead of a redirect
